File: src/main/java/xyz/mattyb/openwest/kotlinprimer/chapter03/Nulls.java
package xyz.mattyb.openwest.kotlinprimer.chapter03;

public class Nulls {
    public static void main(String[] args) {
        final String title = null;

        // will it compile?
        // long salary = null;
        Long salary = null;

        System.out.println("salary before job: " + salary);

        salary = 1_000_000_000_000_000L;

        System.out.println("salary after job: " + salary);

        Employee bob = new Employee();

        Integer headNameLength = null;
        if (bob.getDepartment() != null) {
            if (bob.getDepartment().getHead() != null) {
                if (bob.getDepartment().getHead().getName() != null) {
                    headNameLength = bob.getDepartment().getHead().getName().length();
                }
            }
        }

        System.out.println("head name length: " + headNameLength);
    }
}

File: src/main/java/xyz/mattyb/openwest/kotlinprimer/chapter03/nulls.kt
package xyz.mattyb.openwest.kotlinprimer.chapter03

fun main(args: Array<String>) {
    /**
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     * Don't want to give away how functions work ;-)
     */
    // Will it compile?
    // val salary: Long = null
    var salary: Long? = null

    println("salary before job: $salary")

    // Get a job
    salary = 1_000_000_000_000_000

    println("salary after job: $salary")

    /**
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     */

    val bob = Employee()

    val headNameLength = bob?.department?.head?.name?.length

    println("head name length: $headNameLength")
}